engine/src/voxels/grid.cpp

```diff
@@ -152,27 +157,40 @@ bool VoxelGrid::convert(const VoxelPalette& src, const VoxelPalette& dst, float
     return true;
 }
 
-void cubos::core::data::old::serialize(Serializer& serializer, const VoxelGrid& grid, const char* name)
+bool VoxelGrid::loadFrom(Stream& stream)
 {
-    serializer.beginObject(name);
-    serializer.write(grid.mSize, "size");
-    serializer.write(grid.mIndices, "data");
-    serializer.endObject();
+    uint32_t x, y, z;
+    stream.read(&x, sizeof(uint32_t));
+    stream.read(&y, sizeof(uint32_t));
+    stream.read(&z, sizeof(uint32_t));
```

Comment:
Check if return value of read is really `sizeof(uint32_t)`, and if it isn't, log an error (something like unexpected end of file) and return false.